test/tools/ossfuzz/AbiV2IsabelleFuzzer.cpp

```diff
@@ -43,13 +51,60 @@ DEFINE_PROTO_FUZZER(Contract const& _contract)
 
 	string typeString = converter.isabelleTypeString();
 	string valueString = converter.isabelleValueString();
-	std::cout << typeString << std::endl;
-	std::cout << valueString << std::endl;
 	abicoder::ABICoder coder(abiCoderHeapSize);
-	if (!typeString.empty())
+	if (!typeString.empty() && converter.coderFunction())
 	{
 		auto [encodeStatus, encodedData] = coder.encode(typeString, valueString);
 		solAssert(encodeStatus, "Isabelle abicoder fuzzer: Encoding failed");
+
+		// Raw runtime byte code generated by solidity
+		bytes byteCode;
+		std::string hexEncodedInput;
+
+		try
+		{
+			// Compile contract generated by the proto fuzzer
+			SolidityCompilationFramework solCompilationFramework;
+			std::string contractName = ":C";
+			byteCode = solCompilationFramework.compileContract(contractSource, contractName);
+			Json::Value methodIdentifiers = solCompilationFramework.getMethodIdentifiers();
+			// We always call the second function from the list of alphabetically
```

Comment:
why?

Reply:
Hack: Interface functions are alphabetically sorted. I take advantage of this and the fact that functions generated by the fuzzer are always called the same. The reason this hack is required is because I haven't yet figured out a way to format the interface definition of the function I want to call. This definition changes depending upon the type of variables created by the fuzzer.

Reply:
but why the second? what's the first?

Reply:
Functions in alphabetical order (with hardcoded names but in some cases varargs parameters)

```
checkEncodedCall(bytes4, bytes, uint)
coder_external_calldata(<varargs>)
coder_public_memory(<varargs>)
test()
```

I want to always call `coder_external_calldata` but still haven't figured out how to correctly format the function signature and hence the "second" function hack.

Reply:
Isn't it possible to have a map with the function's name or signature as key, and use that instead?

Reply:
Yeah it should be. So the thing is I need to format struct types as tuples and this is thing that needs careful investigation. For example, the solidity compiler interface contains a map of external function definition to signature but one would need to format the key passed to this map exactly as solidity would store it. The main work here is to format structs as tuples.

For example `function f(S memory)` where `struct S { uint x; bytes y; }` I believe should be formatted as `f((uint,bytes memory))` or something like that. I would need to take care of nested structs.

I felt this shouldn't be a blocker for fuzzing abiv2 especially since we are making it default now and thought if it is better to clean up the hack a bit later.

Wdyt?

Reply:
Ok I see now. Fine with it as is